src/App.js

```diff
@@ -3,49 +3,54 @@ import AddTodoForm from './AddTodoForm';
 import TodoList from './TodoList';
 
 
+
 //Not Easy 
 // Above the App functional component, create a new function named useSemiPersistentState which will be a custom hook
 
 const useSemiPersistentState = () => {
-  const [todoList, setTodoList] = React.useState(
-
-    //Update the default state for todoList to read your "savedTodoList" item from localStorage
+	const [todoList, setTodoList] = React.useState(
+		
+        //Update the default state for todoList to read your "savedTodoList" item from localStorage
 
     //  o	Hint: JSON.parse method  Update your default state to parse the value of the "savedTodoList" item
-    JSON.parse(localStorage.getItem('savedTodoList'))
-  );
-
+        
+        
+        JSON.parse(localStorage.getItem('savedTodoList'))
```

Comment:
If there is nothing in localStorage then your code will return an error on the map function in your ToDoList component. You can fix by creating a fallback value of an array. So, if your JSON.Parse(localStorage.getItem()) comes up undefined then you can make sure to at least have an empty array to pull from.

`JSON.parse(localStorage.getItem('savedTodoList')) || []`

Reply:
The `||` OR value is shorthand for you can use this value if exists OR if not then use this value.